src/java/org/apache/cassandra/cql3/statements/ModificationStatement.java

```diff
@@ -153,6 +172,10 @@ public ModificationStatement(StatementType type,
             modifiedColumns = metadata.regularAndStaticColumns();
 
         this.updatedColumns = modifiedColumns;
+
+        this.pkRestrictedColumns = Columns.from(restrictions.getRestrictions(ColumnMetadata.Kind.PARTITION_KEY).columns());
+        this.clusteringRestrictedColumns = Columns.from(restrictions.getRestrictions(ColumnMetadata.Kind.CLUSTERING).columns());
```

Comment:
Why getting those columns this way? I think you can lookup the column metadata with the column names from `columnValues` map. 

Loop through `metadata.columns()` and progress when the column name matches